packages/modular-scripts/src/utils/getPackageDependencies.ts

```diff
@@ -112,20 +114,24 @@ function parseYarnLockV1(
   lockFile: string,
   deps: Dependency,
 ): Dependency | null {
+  let parsedLockfile: ReturnType<typeof lockfile.parse>;
   try {
-    const parsedLockfile = lockfile.parse(lockFile);
-    return Object.entries(deps).reduce<Record<string, string>>(
-      (acc, [name, version]) => {
-        acc[name] = (parsedLockfile.object as LockFileEntries)[
-          `${name}@${version}`
-        ].version;
-        return acc;
-      },
-      {},
-    );
+    parsedLockfile = lockfile.parse(lockFile);
   } catch (e) {
     return null;
   }
+  return Object.entries(deps).reduce<Record<string, string>>(
+    (acc, [name, version]) => {
+      const resolution = (parsedLockfile.object as LockFileEntries)[
+        `${name}@${version}`
+      ]?.version;
+      if (resolution !== undefined) {
+        acc[name] = resolution;
+      }
+      return acc;
+    },
+    {},
+  );
 }
 
 function parseYarnLockV3(lockFile: string, deps: Dependency): Dependency {
```

Comment:
Not in your PR - but this function may be more readily understood by limiting the loops inside loops by preparing a look-up Map before looping through the lockfile entries:

```typescript
function parseYarnLockV3(lockFile: string, deps: Dependency): Dependency {
  const dependencyArray = Object.entries(deps);
  const npmDependencyList = new Map(
    dependencyArray.map(([name, version]) => [`${name}@npm:${version}`, name])
  );
  const resolved: Dependency = {};

  Object.entries(yaml.load(lockFile) as LockFileEntries).forEach(
    ([name, { version }]) => {
    // Yarn v3 lockfile comes with keys like "'yargs@npm:^15.0.2, yargs@npm:^15.1.0, yargs@npm:^15.3.1, yargs@npm:^15.4.1'" - split them

    const entryDependencies = name.split(', ');
    entryDependencies.some(dep => {
      const npmName = npmDependencyList.get(dep);
      return !!(npmName && (resolved[npmName] = version));
    });
  });

  return resolved;
}
```

Exporting this function from a utility file would then permit unit testing directly without the need to rely on a surrounding integration test.

Reply:
I changed `some` to a `for` loop based on an indication [in a previous PR](https://github.com/jpmorganchase/modular/pull/1793#discussion_r887037489). TBH, I like the pre-calculation of `npmDependencyList` and I find using `reduce` and `some` more compact than explicitly looping and breaking.

+1 for unit testing those functions.

Reply:
Thanks, done (but maintaining the reducer)